### ISSUE TITLE
ngx-nested-ellipsis displaying blank

### ISSUE THREAD

**Issue by bkr32** [lentschi/ngx-nested-ellipsis#5]
After version 1.1.3 ngx-nested-ellipsis displays a blank canvas.

Hi there, i have been using ngx-nested-ellipsis since last december, however i came around and updated my project today and found that ngx-nested-ellipsis would display a blank space rather than the truncated text.

my code is as follows:

```
        <div style="height: 150px">
          <span *nestedEllipsis="true" [innerHTML]="articleData.body| linkify"></span>
        </div>
```

where `linkify` is a custom pipe that replaces special characters within the given `articleData.body`
ie

```
if (t.endsWith(':')) {
          stylizedText += t + `<ul>`;
        }
```

i've noticed two issues here,
firstly the aformentioned issue does not occur when the main div wrapping the span is greater than a certain percent compared to the content,

secondly it does not occur if i do not use the custom pipe, i have gone through the custom pipe code and found no underlying issue, it produces a rich text format that is consumeable by `[innerHTML]`.

Could there be a issue with the white space provided by ngx-nested-ellipsis?

**Comment by lentschi** [comment]
Hm... that's really hard to tell. I'm not sure what you mean by "white space provided by ngx-nested-ellipsis"...? (I don't remember implementing any function adding whitespace, but maybe I just forgot.)

The only valid constellation, in which ngx-nested-ellipsis might not display anything even though the provided text would fit, would be if the html that is to be shortened contains styled elements that break the text flow at an "unfortunate" position. (All ngx-netsted-ellipsis does, is remove DOM elements from the end of the template. But if the css contains styles such as `flex-direction: reverse;` it cannot function properly.)

Maybe you could provide a reproduction case by forking [the StackBlitz sample provided in the README](https://stackblitz.com/github/lentschi/ngx-nested-ellipsis?file=src%2Fapp%2Fapp.component.html), so I can debug the issue?

**Comment by bkr32** [comment]
Unfortunately i'm unable to replicate the issue on StackBlitz, which leads me to believe it must be something in my layout,
`ngx-nested-ellipsis` is nested within `mat-card` which is populated by the parent element using `ngx-masonry`.
stripping all styling from the `mat-card` yielded the same result.

for some inexplicable reason i dont even get the ellipsis on the "blank" renders, rather i get a blank space 
![image](https://user-images.githubusercontent.com/7794407/124895098-e6a45900-dfe4-11eb-8e54-a27a135e3265.png)

in this instance the div is limited to `221px` whereas the expanded content is about `915px`

![image](https://user-images.githubusercontent.com/7794407/124895626-62060a80-dfe5-11eb-9cdd-ca6ba12099a5.png)

in the above instance the div is still `221px`  the same as if it were expanded.

there is no styling applied to the `InnerHTML` just character substitutions ie `<br>` and `<ul>`.

Rolling back to v1.1.2 works for now,

i appoligize for the lack of a stackblitz and if my explanation isn't quite clear

**Comment by lentschi** [comment]
So I take it, it already fails in v1.1.3?

If so, the only substantial change in v1.1.3 was the shift of the resize detection lib from [resize-observer-polyfill](https://www.npmjs.com/package/resize-observer-polyfill) to [@juggle/resize-observer](https://www.npmjs.com/package/@juggle/resize-observer) (required by the angular 11 upgrade) -> To verify that your issue is related to resize detection you could try adding the input `nestedEllipsisResizeDetection="window"` to your `nestedEllipsis` template. If the ellipsis is correctly applied as a result (at least after manually resizing the window), that would mean that the new resize detection ponyfill has trouble working with your layout and/or masonry. 

Are you trying this in a modern browser? ( All modern non-IE browsers should use the [native ResizeDetector](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver#browser_compatibility) - so this would be an odd issue.)
Do you see any js console errors?

**Comment by bkr32** [comment]
Thank you for your prompt reply,

Yes, last working version is 1.1.2.

i attempted the above fix 
```
        <div style="height: 180px">
          <ng-template nestedEllipsis="true" nestedEllipsisResizeDetection="window">
            <p [innerHTML]="articleData.body | linkify"></p>
          </ng-template>
        </div>
```

but unfortunately it yielded the same result.

i have tested this on Firefox 87 as well as Chrome 91

**Comment by chloe-1903** [comment]
I also noticed that I have the same blank space when I have several paragraphs in my text. Something like [innerHtml]="'<p>toto</p> <p>titi</p>'". Even if I wrap that into a div : <div> <p>toto</p> <p>titi</p> </div>

**Comment by lentschi** [comment]
@bkr32 Fixed in `ngx-nested-ellipsis@1.1.5` - thanks for providing the StackBlitz - it helped resolving the issue, which has existed since a2acfaa622730d (`ngx-nested-ellipsis@1.1.3`)

As @chloe-1903 correctly pointed out, it also happened with simple HTML and without masonry.

**Comment by bkr32** [comment]
Thanks for the swift fix @lentschi glad I was able to help, given the multiple nested components I'm afraid I missed out on the multi paragraph being a cause
